switchapi/es2k/switch_lag.c

```diff
@@ -0,0 +1,472 @@
+  CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+  if (!SWITCH_LAG_MEMBER_HANDLE(lag_member_h)) {
+    status = SWITCH_STATUS_INVALID_PARAMETER;
+    krnlmon_log_error(
+        "lag get: Invalid lag member handle on device %d, "
+        "lag member handle 0x%lx: "
+        "error: %s\n",
+        device, lag_member_h, switch_error_to_string(status));
+    return status;
+  }
+  status = switch_lag_member_get(device, lag_member_h, &lag_member_info);
+  CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+  if (lag_info->active_lag_member != 0) {
+    // delete rx path
+    status = switch_pd_rx_lag_table_entry(device, lag_info, false);
+    if (status != SWITCH_STATUS_SUCCESS) {
+      krnlmon_log_error(
+          "Failed to delete rx lag table entry on device %d: "
+          ",error: %s\n",
+          device, switch_error_to_string(status));
+      return status;
+    }
+
+    // insert lag member
+    status = SWITCH_LIST_INSERT(&(lag_info->lag_members),
+                                &(lag_member_info->node), lag_member_info);
+    SWITCH_ASSERT(status == SWITCH_STATUS_SUCCESS);
+
+    // create rx path
+    status = switch_pd_rx_lag_table_entry(device, lag_info, true);
+    if (status != SWITCH_STATUS_SUCCESS) {
+      krnlmon_log_error(
+          "Failed to create rx lag table entry on device %d: "
+          ",error: %s\n",
+          device, switch_error_to_string(status));
+      return status;
+    }
+  } else {
+    // update lag members list with the lag_member_h
+    status = SWITCH_LIST_INSERT(&(lag_info->lag_members),
+                                &(lag_member_info->node), lag_member_info);
+    SWITCH_ASSERT(status == SWITCH_STATUS_SUCCESS);
+  }
+
+  return status;
+}
+
+/*
+ * Routine Description:
+ *   @brief Get LAG attributes
+ *
+ * Arguments:
+ *   @param[in] device - device
+ *   @param[in] lag_h - LAG handle
+ *   @param[in] lag_flags - LAG flags
+ *   @param[out] api_lag_info - LAG info struct
+ *
+ * Return Values:
+ *    @return  SWITCH_STATUS_SUCCESS on success
+ *             Failure status code on error
+ */
+switch_status_t switch_api_lag_attribute_get(
+    const switch_device_t device, const switch_handle_t lag_h,
+    const switch_uint64_t lag_flags, switch_api_lag_info_t* api_lag_info) {
+  switch_lag_info_t* lag_info = NULL;
+  switch_status_t status = SWITCH_STATUS_SUCCESS;
+
+  if (!SWITCH_LAG_HANDLE(lag_h)) {
+    status = SWITCH_STATUS_INVALID_PARAMETER;
+    krnlmon_log_error(
+        "lag attribute get: Invalid lag handle on device %d, "
+        "lag handle 0x%lx: "
+        "error: %s\n",
+        device, lag_h, switch_error_to_string(status));
+    return status;
+  }
+
+  status = switch_lag_get(device, lag_h, &lag_info);
+  CHECK_RET(status != SWITCH_STATUS_SUCCESS, status);
+
+  *api_lag_info = lag_info->api_lag_info;
+
+  return status;
+}
+
+/*
+ * Routine Description:
+ *   @brief Program the Tx and Rx LAG tables
+ *
+ * Arguments:
+ *   @param[in] lag_h : LAG handle
```

Comment:
please add this too

@param[in] device - device

Reply:
Added.